crypto/rsa/rsa_ameth.c

```diff
@@ -1148,24 +1148,21 @@ static int rsa_pkey_export_to(const EVP_PKEY *from, void *to_keydata,
         for (i = 0; i < numprimes; i++) {
             const BIGNUM *num = sk_BIGNUM_const_value(primes, i);
 
-            if (!ossl_param_bld_push_BN(&tmpl, OSSL_PKEY_PARAM_RSA_FACTOR,
-                                        num))
+            if (!ossl_param_bld_push_BN(&tmpl, rsa_factor_names[i], num))
```

Comment:
I'm lost, where does that array come from?

Reply:
providers/implementations/keymgmt/kmgmt_util.c

Reply:
I finally found it... in the deep recesses of provider code. I'm not sure I agree with this. We *do* cheat some, but this brings it to a new level...

Reply:
At what level should I store them? crypto??

Reply:
You should avoid shared storage if this sort. The proper way of finding out is with a gettable function and pilfering through the returned OSSL_PARAM array, but before you say it, that's problematic too since there's other stuff there as well.

This is actually a reason why I went with multiple instances of the same name. I was in exactly the rabbit hole you're finding yourself in a while ago, 'cause I also considered numbered names back when I added MPRSA support.

If you really want to do this, yeah I would have those arrays in a separate source file in crypto/rsa, and make sure that file gets added to the source of libcrypto and libfips.a

Reply:
thanks - will do

Reply:
Generally speaking, I believe that code that is shared between crypto/FOO/FOO_ameth.c and whatever within providers/implementations should go into crypto/FOO. At some point, when all legacy backends are gone, we can move it all to the providers tree.